src/emtmlibpy/emtmlibpy.py

```diff
@@ -0,0 +1,795 @@
+        self.b_compound_length = b_compound_length
+        self.d_imx1_left = d_imx1_left
+        self.d_imy1_left = d_imy1_left
+        self.d_imx1_right = d_imx1_right
+        self.d_imy1_right = d_imy1_right
+        self.d_imx2_left = d_imx2_left
+        self.d_imy2_left = d_imy2_left
+        self.d_imx2_right = d_imx2_right
+        self.d_imy2_right = d_imy2_right
+        self.d_length = d_length
+        self.d_precision = d_precision
+        self.d_rms = d_rms
+        self.d_range = d_range
+        self.d_direction = d_direction
+        self.dx_mid = dx_mid
+        self.dy_mid = dy_mid
+        self.dz_mid = dz_mid
+        self.str_family = str_family
+        self.str_genus = str_genus
+        self.str_species = str_species
+        self.str_code = str_code
+        self.str_number = str_number
+        self.str_stage = str_stage
+        self.str_activity = str_activity
+        self.str_comment = str_comment
+        self.str_att_9 = str_att_9
+        self.str_att_10 = str_att_10
+
+
+class TmPointData(ctypes.Structure):
+    """
+    Point structures used by libEMTLib.so from SeaGIS
+    """
+    _fields_ = [
+        ('str_filename', ctypes.c_char * EMTM_MAX_CHARS),
+        ('n_frame', ctypes.c_int),
+        ('d_time_mins', ctypes.c_double),
+        ('d_image_row', ctypes.c_double),
+        ('d_image_col', ctypes.c_double),
+        ('str_att_1', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_2', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_3', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_4', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_5', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_6', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_7', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_8', ctypes.c_char * EMTM_MAX_CHARS)
+    ]
+
+    def __int__(self,
+                str_filename,
+                n_frame,
+                d_time_mins,
+                d_image_row,
+                d_image_col,
+                str_att_1, str_att_2, str_att_3, str_att_4, str_att_5, str_att_6, str_att_7, str_att_8):
+        super().__init__()
+        self.str_filename = str_filename
+        self.n_frame = n_frame
+        self.d_time_mins = d_time_mins
+        self.d_image_row = d_image_row
+        self.d_image_col = d_image_col
+        self.str_att_1 = str_att_1
+        self.str_att_2 = str_att_2
+        self.str_att_3 = str_att_3
+        self.str_att_4 = str_att_4
+        self.str_att_5 = str_att_5
+        self.str_att_6 = str_att_6
+        self.str_att_7 = str_att_7
+        self.str_att_8 = str_att_8
+
+
+def emtm_version() -> tuple[int, int]:
+    """
+    Return the version number of libStereoLibLX
+    :return: (major, minor)
+    """
+
+    pminor = ctypes.POINTER(ctypes.c_int)
+    pmajor = ctypes.POINTER(ctypes.c_int)
+    libc.EMTMVersion.argtypes = [pminor, pmajor]
+
+    minor = ctypes.c_int(0)
+    major = ctypes.c_int(0)
+
+    libc.EMTMVersion(major, minor)
```

Comment:
Should you not be passing the pointers in here?

Reply:
Feel like this always returns (0, 0)

Reply:
interesting! maybe they get passed by reference automatically? I'm not familiar with ctypes so possible I've misunderstood how this works

Reply:
Explicitly passing by ref now.